x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/analyzer_preview.tsx

```diff
@@ -78,17 +88,34 @@ export const AnalyzerPreview: React.FC = () => {
 
   const showAnalyzerTree = items && items.length > 0 && !error;
 
-  return loading ? (
-    <EuiSkeletonText
-      data-test-subj={ANALYZER_PREVIEW_LOADING_TEST_ID}
-      contentAriaLabel={i18n.translate(
-        'xpack.securitySolution.flyout.right.visualizations.analyzerPreview.loadingAriaLabel',
-        {
-          defaultMessage: 'analyzer preview',
-        }
-      )}
-    />
-  ) : showAnalyzerTree ? (
+  if (dataViewLoading || dataLoading) {
+    return (
+      <EuiSkeletonText
+        data-test-subj={ANALYZER_PREVIEW_LOADING_TEST_ID}
+        contentAriaLabel={i18n.translate(
+          'xpack.securitySolution.flyout.right.visualizations.analyzerPreview.loadingAriaLabel',
+          {
+            defaultMessage: 'analyzer preview',
+          }
+        )}
+      />
+    );
+  }
+
+  if (
+    status === 'error' ||
+    (status === 'ready' && !dataView.hasMatchedIndices()) ||
+    !showAnalyzerTree
+  ) {
+    return (
+      <FormattedMessage
+        id="xpack.securitySolution.flyout.right.visualizations.analyzerPreview.errorDescription"
+        defaultMessage="An error is preventing this alert from being analyzed."
+      />
+    );
+  }
+
```

Comment:
NIT: I think it would be nice to show different errors for data-view loading and analyser requests like you did in the analyze_graph component.

I think it would be useful especially if we start getting SDHs about this, so we can triage the issue quicker ☺️

What do you think?